devito/function_manager.py

```diff
@@ -19,6 +20,11 @@ def __init__(self, function_descriptors, mic_flag=False, openmp=False):
         if openmp:
             self.libraries = self.libraries + ['omp.h']
 
+        for fd in self.function_descriptors:  # appends main at function if flag is set to true
+            if fd.add_main_at_function:
+                self._append_main_at_function(fd)
+                break
```

Comment:
This can be replaced by a list comprehension and a call to [any](https://docs.python.org/2/library/functions.html#any)